relay-event-normalization/src/validation.rs

```diff
@@ -11,9 +11,24 @@ use relay_protocol::{Annotated, ErrorKind, Meta};
 
 use crate::{ClockDriftProcessor, TimestampProcessor};
 
-/// Configuration for [`validate_transaction`].
+/// Configuration for [`validate_event`].
 #[derive(Debug, Default)]
-pub struct TransactionValidationConfig {
+pub struct EventValidationConfig {
+    /// The time at which the event was received in this Relay.
+    ///
+    /// This timestamp is persisted into the event payload.
+    pub received_at: Option<DateTime<Utc>>,
+
+    /// The maximum amount of seconds an event can be dated in the past.
+    ///
+    /// If the event's timestamp is older, the received timestamp is assumed.
+    pub max_secs_in_past: Option<i64>,
+
+    /// The maximum amount of seconds an event can be predated into the future.
+    ///
+    /// If the event's timestamp lies further into the future, the received timestamp is assumed.
+    pub max_secs_in_future: Option<i64>,
```

Comment:
This PR surfaces an inconsistency:

1. For timestamp correction we use two integers but for transactions we use a range.
2. We correct timestamps that are very old (30 days), but then we still dismiss transactions when they are older than 5 days. Should we correct transaction timestamps to 5 days instead?

Reply:
Treating this as resolved: 1. needs optional timestamps, 2. is a business decision.